pytorch_lightning/trainer/train_loop_mixin.py

```diff
@@ -213,7 +214,15 @@ def train(self):
             # update LR schedulers
             if self.lr_schedulers is not None:
                 for lr_scheduler in self.lr_schedulers:
-                    lr_scheduler.step(self.current_epoch)
+                    lr_scheduler.step(epoch=self.current_epoch)
+            if self.reduce_lr_on_plateau_scheduler is not None:
+                val_loss = self.callback_metrics.get('val_loss')
```

Comment:
this seems very specific. does it only need to work with val_loss?

Reply:
I think it could be any validation metric in theory, but how do we let the user pass it in? Perhaps via a dedicated dict entry in the [validation_end](https://williamfalcon.github.io/pytorch-lightning/LightningModule/RequiredTrainerInterface/#validation_end) output similar to "log" and "progress_bar"? It's one more thing the user needs to remember, but maybe its fine since this lr_scheduler is optional and it should be mentioned in the docs.

Reply:
yeah, good option. let’s do it in a separate PR?